lib/client/indexer/indexerClient.ts

```diff
@@ -12,12 +15,51 @@ export class FutarchyIndexerClient implements FutarchyClient {
   public proposals: FutarchyIndexerProposalsClient;
   public balances: FutarchyIndexerBalancesClient;
   public markets: FutarchyIndexerMarketsClient;
+  private protocolMap: Map<string, FutarchyProtocol>;
 
-  constructor() {
-    //TODO: implement all this
-    this.daos = new FutarchyIndexerDaoClient();
-    this.proposals = new FutarchyIndexerProposalsClient();
-    this.balances = new FutarchyIndexerBalancesClient();
-    this.markets = new FutarchyIndexerMarketsClient();
+  constructor(
+    rpcClient: FutarchyRPCClient,
+    indexerURL: string,
+    indexerApiKey: string
+  ) {
+    if (indexerURL === "" || indexerApiKey === "")
+      throw "missing url or api key for indexer client";
+
+    // TODO how can we batch these queries??
+    const options = {
+      url: indexerURL,
+      headers: {
+        // TODO this admin secret in the request... should not be deployed to staging probably
```

Comment:
correct, we likely want a jwt token, however not sure what may work out of the box with hasura @decartesol may know. Otherwise setting up an haproxy isn't ideal but maybe...